app/models/host/managed.rb

```diff
@@ -196,6 +196,7 @@ class Jail < ::Safemode::Jail
     validate :short_name_periods
     before_validation :set_compute_attributes, :on => :create, :if => Proc.new { compute_attributes_empty? }
     validate :check_if_provision_method_changed, :on => :update, :if => Proc.new { |host| host.managed }
+    validates :uuid, uniqueness: { :scope => :compute_resource_id }
```

Comment:
its a good question if it should be unique across vs a specific compute resource (e.g. if someone moved a vm? or created a duplicate compute resource with different creds for example)?